collator/src/storage/transaction.rs

```diff
@@ -66,39 +65,21 @@ impl InternalQueueTransaction {
         self.batch.put_cf(&cf, key.to_vec(), self.buffer.as_slice());
     }
 
+    /// Updates commit pointers in message queue.
+    /// ATTENTION! Overrides old value without checks. Should validate the new value in the calling code.
     pub fn commit_messages(
         &mut self,
-        commit_pointers: &FastHashMap<ShardIdent, (QueueKey, u32)>,
+        commit_pointers: FastHashMap<ShardIdent, (QueueKey, u32)>,
     ) -> Result<()> {
         let commit_pointers_cf = self.db.internal_message_commit_pointer.cf();
 
-        for (&shard_ident, (queue_key, seqno)) in commit_pointers.iter() {
+        for (shard_ident, (queue_key, seqno)) in commit_pointers {
             let key = CommitPointerKey { shard_ident }.to_vec();
 
-            // Get the old value if it exists
```

Comment:
Why do you removed this logic but didn't add to caller in same commit? Won't this change the logic of the node's work and cause errors?